packages/flutter/lib/src/material/material_state.dart

```diff
@@ -309,26 +301,28 @@ abstract class MaterialStateOutlineInputBorder extends OutlineInputBorder implem
   /// Creates a [MaterialStateOutlineInputBorder] from a [MaterialPropertyResolver<InputBorder>]
   /// callback function.
   ///
-  /// If used as a regular input border, the border resolved in the default state (the
-  /// empty set of states) will be used.
```

Comment:
I agree that we should lead with directing people to not use these classes outside of their intended use cases. Only telling them how they work when used on a regular property is almost directing people to hack. But it might be worth leaving in a description of their behavior when used on a property that doesn't handle them may be useful for debugging purposes.

Reply:
> may be useful for debugging purposes

Excellent point, will update 👍